### PR TITLE
Mooncake updates

### PR DESCRIPTION
**Pull request by faister**
## Mooncake updates

1. Tracked in issue #867  

## This PR fixes/adds/changes/removes

1. Updated [DIY instructions](https://github.com/Azure/Enterprise-Scale/blob/main/eslzArm/README-AzureChina.md) for deploying Enterprise-Scale in Azure China with:
- Additional details of some deployment steps
- Microsoft Defender for Cloud configuration policy set definition and policy assignment specific to Azure China
- Differentiate between Az Vm Backup policy assignment for identity management group, and landing zone management group in the DIY guidance
2. The following policy definitions for Microsoft Defender for Cloud configurations are not available as built-in in Azure China. The policy set definition will be updated as when these policy definitions are available:
- defenderForOssDb, defenderForSqlServerVirtualMachines, defenderForAppServices, defenderForAppServices, defenderForStorageAccounts, defenderForKubernetesService, defenderForKeyVaults, defenderForDns, defenderForArm
- defenderForContainerRegistry - this built-in policy has been deprecated (see below, I saw this is also the case for Azure global regions)
- 
image

### Breaking Changes

None.

## Testing Evidence
image

## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Enterprise-Scale/pulls)
- [x] Associated it with relevant [issues](https://github.com/Azure/Enterprise-Scale/issues), for tracking and closure.
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Enterprise-Scale/tree/main)
- [x] Performed testing and provided evidence.
- [x] Updated relevant and associated documentation.
- [x] Updated the ["What's New?"](https://github.com/Azure/Enterprise-Scale/wiki/Whats-new) wiki page (located: `/docs/wiki/whats-new.md`)

### CODE REVIEW

**Comment by jtracey93** [comment]
Nice work @faister 👍

One question, did the previous custom policies for ASC/MDFC work for all the supported SKUs in ASC/MDFC before? And if so, is it worth "plugging the gap" with those for those that aren't available as built in for now? 

Interested in your thoughts on this 👍

File: eslzArm/README-AzureChina.md
Comment (krnese) [inline eslzArm/README-AzureChina.md:31]: Can we rather say this is due to a known replication delay, as the dependency graph is indeed respected, but due to how traffic is being routed the replication may not be completed when trying to GET the definitions that just got created?

Comment (krnese) [reply eslzArm/README-AzureChina.md:31]: As documented @ https://github.com/Azure/Enterprise-Scale/blob/main/docs/EnterpriseScale-Known-Issues.md#deploying-the-reference-implementation-fails-due-to-policy--cannot-be-found-404

**Comment by faister** [comment]
> Nice work @faister Fai Lai FTE 👍
> 
> One question, did the previous custom policies for ASC/MDFC work for all the supported SKUs in ASC/MDFC before? And if so, is it worth "plugging the gap" with those for those that aren't available as built in for now?
> 
> Interested in your thoughts on this 👍

Thanks mate. If those Defender for Cloud policies (defenderForOssDb, defenderForSqlServerVirtualMachines, defenderForAppServices, defenderForAppServices, defenderForStorageAccounts, defenderForKubernetesService, defenderForKeyVaults, defenderForDns, defenderForArm) were custom to begin with, then yes I'd be inclined to "plug the gap" until they become built-in policies in Mooncake. I'd taken this stance in the previous Mooncake PR. But since these were new built-in policies introduced in Azure global regions, but not yet in Mooncake, let's wait for them to land.

The one and only exception is that I cannot find any equivalent defender policy for container registries to replace the deprecated built-in policy, defenderForContainerRegistry,  /providers/Microsoft.Authorization/policyDefinitions/d3d1e68e-49d4-4b56-acff-93cef644b432

File: eslzArm/README-AzureChina.md
Comment (faister) [reply eslzArm/README-AzureChina.md]: @krnese yeah that makes sense, I have reworded the explanation and committed it below

**Comment by jtracey93** [comment]
> > Nice work @faister
> >             
> >             Fai Lai
> >             FTE Fai Lai FTE 👍
> > One question, did the previous custom policies for ASC/MDFC work for all the supported SKUs in ASC/MDFC before? And if so, is it worth "plugging the gap" with those for those that aren't available as built in for now?
> > Interested in your thoughts on this 👍
> 
> Thanks mate. If those Defender for Cloud policies (defenderForOssDb, defenderForSqlServerVirtualMachines, defenderForAppServices, defenderForAppServices, defenderForStorageAccounts, defenderForKubernetesService, defenderForKeyVaults, defenderForDns, defenderForArm) were custom to begin with, then yes I'd be inclined to "plug the gap" until they become built-in policies in Mooncake. I'd taken this stance in the previous Mooncake PR. But since these were new built-in policies introduced in Azure global regions, but not yet in Mooncake, let's wait for them to land.
> 
> The one and only exception is that I cannot find any equivalent defender policy for container registries to replace the deprecated built-in policy, defenderForContainerRegistry, /providers/Microsoft.Authorization/policyDefinitions/d3d1e68e-49d4-4b56-acff-93cef644b432

So for clarity, we are not going to plug the gap where the new built-ins do not cover Mooncake. Instead, we are going to await them as they rollout and add them to our initiative?

As for the ACR policies, this is likely due to the changes coming for Defender for Cloud, see [Container security features to be grouped under Defender for Containers](https://docs.microsoft.com/en-us/azure/defender-for-cloud/upcoming-changes#container-security-features-to-be-grouped-under-defender-for-containers)

**Comment by jtracey93** [comment]
Just on this @faister i have now created issue #874 so we can track this ACR change separately

**Comment by faister** [comment]
> > > Nice work @faister
> > >             
> > >             Fai Lai
> > >             FTE
> > > ```
> > >         Fai Lai
> > >         FTE Fai Lai FTE 👍
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > One question, did the previous custom policies for ASC/MDFC work for all the supported SKUs in ASC/MDFC before? And if so, is it worth "plugging the gap" with those for those that aren't available as built in for now?
> > > Interested in your thoughts on this 👍
> > 
> > 
> > Thanks mate. If those Defender for Cloud policies (defenderForOssDb, defenderForSqlServerVirtualMachines, defenderForAppServices, defenderForAppServices, defenderForStorageAccounts, defenderForKubernetesService, defenderForKeyVaults, defenderForDns, defenderForArm) were custom to begin with, then yes I'd be inclined to "plug the gap" until they become built-in policies in Mooncake. I'd taken this stance in the previous Mooncake PR. But since these were new built-in policies introduced in Azure global regions, but not yet in Mooncake, let's wait for them to land.
> > The one and only exception is that I cannot find any equivalent defender policy for container registries to replace the deprecated built-in policy, defenderForContainerRegistry, /providers/Microsoft.Authorization/policyDefinitions/d3d1e68e-49d4-4b56-acff-93cef644b432
> 
> So for clarity, we are not going to plug the gap where the new built-ins do not cover Mooncake. Instead, we are going to await them as they rollout and add them to our initiative?
> 
> As for the ACR policies, this is likely due to the changes coming for Defender for Cloud, see [Container security features to be grouped under Defender for Containers](https://docs.microsoft.com/en-us/azure/defender-for-cloud/upcoming-changes#container-security-features-to-be-grouped-under-defender-for-containers)
> 
> Which looks right as the AKS & ACR policies are both now showing as deprecated and on the same day this new policy appeared: [Configure Microsoft Defender for Containers to be enabled](https://www.azadvertizer.net/azpolicyadvertizer/c9ddb292-b203-4738-aead-18e2716e858f.html)

As per our conversation, I will create an issue to track these new MSD built-ins not available in Mooncake to make sure that we don't forget about them, and have Jeff follow up with the PG. Since Mooncake DIY deployment is broken now, the priority is to get this back up and running, appreciate both you and @krnese to approve this if there is nothing else too pressing.

**Review by jtracey93 — APPROVED** [review]

**Review by krnese — APPROVED** [review]

**Pull request merged**
